sql/parse/parse.go

```diff
@@ -525,6 +525,19 @@ func convertShow(ctx *sql.Context, s *sqlparser.Show, query string) (sql.Node, e
 			if err != nil {
 				return nil, err
 			}
+			// TODO: once collations are properly implemented, we should better be able to handle utf8 -> utf8mb3 comparisons as they're aliases
+			filterExpr, err = expression.TransformUp(ctx, filterExpr, func(expr sql.Expression) (sql.Expression, error) {
```

Comment:
engine test for this?

Reply:
This is caught by one of the modified engine tests, in particular the `IN (<collations>)` one.